sources/engine/Stride.Physics/Simulation.cs

```diff
@@ -1045,8 +1057,25 @@ internal void Simulate(float deltaTime)
 
             SimulationProfiler = Profiler.Begin(PhysicsProfilingKeys.SimulationProfilingKey);
 
-            if (discreteDynamicsWorld != null) discreteDynamicsWorld.StepSimulation(deltaTime, MaxSubSteps, FixedTimeStep);
-            else collisionWorld.PerformDiscreteCollisionDetection();
+            if (discreteDynamicsWorld != null)
+            {
+                carriedDelta += deltaTime;
+                for (; carriedDelta > FixedTimeStep; )
```

Comment:
nit: you can use a `while` loop here for better readability

Reply:
That was the idea initially, but forgot about it while testing, ended up extending it into a cleaner for loop instead